internal/pkg/client/cache/dir.go

```diff
@@ -38,6 +38,13 @@ const (
 	CacheDir = "cache"
 )
 
+// Context describes the context in which a new handle is created, as defined
+// by the user through command flags and environment variables.
+type Context struct {
```

Comment:
I have a knee-jerk reaction to the name of this struct just because of the prevalence of the https://golang.org/pkg/context/ package to manage go routine lifetimes. What are your thoughts on renaming this to `Config`?